app/src/main/java/de/westnordost/streetcomplete/quests/lanes/AddLanes.kt

```diff
@@ -0,0 +1,72 @@
+package de.westnordost.streetcomplete.quests.lanes
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.osmquest.OsmFilterQuestType
+
+class AddLanes : OsmFilterQuestType<LanesAnswer>() {
+
+    override val elementFilter = """
+        ways with
+          highway ~ ${ROADS_WITH_LANES.joinToString("|")}
+          and (!lanes or lanes !~ [1-9][0-9]*)
+    """
+    override val commitMessage = "Add road lanes"
+    override val wikiLink = "Key:lanes"
+    override val icon = R.drawable.ic_quest_street_lanes
+    override val isSplitWayEnabled = true
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_lanes_title
+
+    override fun createForm() = AddLanesForm()
+
+    override fun applyAnswerTo(answer: LanesAnswer, changes: StringMapChangesBuilder) {
+
+        changes.addOrModify("lanes", answer.total.toString())
+
+        val isUnmarked = answer !is UnmarkedLanes
+        // if there is just one lane, the information whether it is marked or not is irrelevant
+        // (if there are no more than one lane, there are no markings to separate them)
+        if (answer.total == 1) changes.deleteIfExists("lane_markings")
+        else if (isUnmarked)   changes.modifyIfExists("lane_markings", "yes")
+        else                   changes.addOrModify("lane_markings", "no")
+
+        val hasCenterLeftTurnLane = answer is MarkedLanesSides && answer.centerLeftTurnLane
+        if (hasCenterLeftTurnLane) {
+            changes.addOrModify("lanes:both_ways", "1")
+            changes.addOrModify("turn:lanes:both_ways","left")
+        } else {
+            changes.deleteIfExists("lanes:both_ways")
+            changes.deleteIfExists("turn:lanes:both_ways")
+        }
+
+        when(answer) {
+            is MarkedLanes -> {
+                if (answer.count == 1) {
+                    changes.deleteIfExists("lanes:forward")
+                    changes.deleteIfExists("lanes:backward")
+                } else {
+                    changes.modifyIfExists("lanes:forward", (answer.count / 2).toString())
+                    changes.modifyIfExists("lanes:backward", (answer.count / 2).toString())
+                }
+            }
+            is UnmarkedLanes -> {
+                changes.deleteIfExists("lanes:forward")
+                changes.deleteIfExists("lanes:backward")
+            }
+            is MarkedLanesSides -> {
+                changes.addOrModify("lanes:forward", answer.forward.toString())
+                changes.addOrModify("lanes:backward", answer.backward.toString())
+            }
+        }
+    }
+
+    companion object {
+        private val ROADS_WITH_LANES = listOf(
```

Comment:
I was unsure about _links but for motorway_link it seems that asking will be useful: http://overpass-turbo.eu/s/107h (1/3 of cases where `lanes` is tagged motorway link is larger than smallest possible one)


For primary link about 1/5: http://overpass-turbo.eu/s/107i

Yes, it is possible that every single interesting `lane` value was tagged for them in some regions - but even if in some regions asking for lane count on links is not useful anymore then it still will not be a spam, links are not appearing so often